edgexfoundry/device-sdk-go
# feat: Implement URI for Profile, Device, & Provision Watcher

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

**Non-Secure testing**
1. pull the latest code and build device simple
2. run the non-secure edgex stack
3. run device-simple with its standard configuration
4. stop device-simple 
5. modify the configuration to point to a custom index.json file (created in internal/uri-test-files/*/index.json)
6. run device-simple and ensure that the index.json works without errors

** Secure testing**
1. create a private repo
2. run the secure edgex stack
3. use both configuration options - file and URI
4. run and ensure device-simple works

## New Dependency Instructions (If applicable)
